src/models/cost_functions/ImportExportCost.jl

```diff
@@ -0,0 +1,259 @@
+        Nothing,
+        TimeSeriesKey,  # piecewise step data
+        CostCurve{PiecewiseIncrementalCurve},
+    }
+    "Sell Price Curves data to export power, which can be a time series of `PiecewiseStepData` or a
+    [`CostCurve`](@ref) of [`PiecewiseIncrementalCurve`](@ref)"
+    export_offer_curves::Union{
+        Nothing,
+        TimeSeriesKey,
+        CostCurve{PiecewiseIncrementalCurve},
+    }
+    "Bids to buy or sell ancillary services in the interconnection"
+    ancillary_service_offers::Vector{Service}
+    "TODO docstring"
+    energy_import_weekly_limit::Float64
+    "TODO docstring"
+    energy_export_weekly_limit::Float64
+end
+
+ImportExportCost(;
+    import_offer_curves = nothing,
+    export_offer_curves = nothing,
+    ancillary_service_offers = Vector{Service}(),
+    energy_import_weekly_limit = 0.0,
+    energy_export_weekly_limit = 0.0,
+) = ImportExportCost(
+    import_offer_curves,
+    export_offer_curves,
+    ancillary_service_offers,
+    energy_import_weekly_limit,
+    energy_export_weekly_limit,
+)
+
+# Constructor for demo purposes; non-functional.
+function ImportExportCost(::Nothing)
+    ImportExportCost()
+end
+
+"""Get [`ImportExportCost`](@ref) `import_offer_curves`."""
+get_import_offer_curves(value::ImportExportCost) = value.import_offer_curves
+"""Get [`ImportExportCost`](@ref) `export_offer_curves`."""
+get_export_offer_curves(value::ImportExportCost) = value.export_offer_curves
+"""Get [`ImportExportCost`](@ref) `ancillary_service_offers`."""
+get_ancillary_service_offers(value::ImportExportCost) = value.ancillary_service_offers
+"""Get [`ImportExportCost`](@ref) `energy_import_weekly_limit`."""
+get_energy_import_weekly_limit(value::ImportExportCost) = value.energy_import_weekly_limit
+"""Get [`ImportExportCost`](@ref) `energy_export_weekly_limits`."""
+get_energy_export_weekly_limit(value::ImportExportCost) = value.energy_export_weekly_limit
+
+"""Set [`ImportExportCost`](@ref) `import_offer_curves`."""
+set_import_offer_curves!(value::ImportExportCost, val) =
+    value.import_offer_curves = val
+"""Set [`ImportExportCost`](@ref) `export_offer_curves`."""
+set_export_offer_curves!(value::ImportExportCost, val) =
+    value.export_offer_curves = val
+"""Set [`ImportExportCost`](@ref) `ancillary_service_offers`."""
+set_ancillary_service_offers!(value::ImportExportCost, val) =
+    value.ancillary_service_offers = val
+"""Get [`ImportExportCost`](@ref) `energy_import_weekly_limit`."""
+set_energy_import_weekly_limit!(value::ImportExportCost, val) =
+    value.energy_import_weekly_limit = val
+"""Get [`ImportExportCost`](@ref) `energy_export_weekly_limits`."""
+set_energy_export_weekly_limit!(value::ImportExportCost, val) =
+    value.energy_export_weekly_limit = val
+
+function is_import_export_curve(curve::ProductionVariableCostCurve)
+    return (curve isa CostCurve{PiecewiseIncrementalCurve}) &&
+           iszero(get_initial_input(get_value_curve(curve)))
```

Comment:
We should probably also check `iszero(get_input_at_zero(get_value_curve(curve)))` and `iszero(first(get_x_coords(get_value_curve(curve))))`